File: leetCode/fukun/101-200/lc_107.js
function TreeNode(val) {
    this.val = val;
    this.left = this.right = null;
}

const levelOrderBottom = function (root) {
    let output = [];
    const doPush = function (node, level) {
        if (node === null) return;
        output[level] ? output[level].push(node.val) : output.push([node.val]);
        doPush(node.left, level + 1);
        doPush(node.right, level + 1);
    };
    doPush(root, 0);
    return output.reverse();
};

let t = new TreeNode(1);
t.left = new TreeNode(2);
t.right = new TreeNode(2);

console.log(levelOrderBottom(t));

File: leetCode/lyc/e/lc_107.java
/*Binary Tree Level Order Traversall II*/
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrderBottom(TreeNode root) {
       LinkedList<List<Integer>> list = new LinkedList<>();
        
        if(root == null){
            return list;
        }
        
        Queue<TreeNode> queue = new LinkedList<>();
        queue.add(root);
        
        while(!queue.isEmpty()){
            int sum = queue.size();
            List<Integer> temp = new ArrayList<>();
            while(sum > 0){
                TreeNode node = queue.poll();
                temp.add(node.val);
                if(node.left != null){
                    queue.add(node.left);
                }
                if(node.right != null){
                    queue.add(node.right);
                }
                sum--;
            }
            list.addFirst(temp);
        }
        return list; 
    }
}